Sources/NIOFileSystem/Internal/System Calls/Syscalls.swift

```diff
@@ -452,21 +452,12 @@ internal func libc_confstr(
 #endif
 
 /// fts(3)
-#if os(Android)
-internal func libc_fts_open(
-    _ path: [UnsafeMutablePointer<CInterop.PlatformChar>],
-    _ options: CInt
-) -> UnsafeMutablePointer<CInterop.FTS> {
-    fts_open(path, options, nil)!
-}
-#else
 internal func libc_fts_open(
     _ path: [UnsafeMutablePointer<CInterop.PlatformChar>?],
```

Comment:
This was [incorrectly marked as `_Nonnull` in recent Android NDKs, which will be fixed in the upcoming NDK 28](https://android.googlesource.com/platform/bionic/+/da81ec4d1cbd0279014feb60535bf38defcd9346%5E%21/#F0), but in the meantime, Alex added an API note to fix it for 6.1, [which is about to be backported to 6.0 also](https://github.com/swiftlang/swift/pull/77626/files#diff-80ad50e239d0b1f4236139c396a2f6e9e53dd96c3726de022e4efa7b4b2e1addR4).

Once it is, the alternate Android definition above is unneeded for Swift 6, so I'm simply doing the force-unwrap for Android here instead.